java/test/jmri/jmrix/tmcc/serialmon/SerialMonFrameTest.java

```diff
@@ -23,20 +17,19 @@
 public class SerialMonFrameTest {
 
     @Test
+    @DisabledIfHeadless
     public void testCreateAndShow() {
-        Assume.assumeFalse(GraphicsEnvironment.isHeadless());
+
         TmccSystemConnectionMemo memo = new TmccSystemConnectionMemo("T", "TMCC via Serial");
-        memo.setTrafficController(new SerialTrafficController(memo));
+        memo.setTrafficController(new SerialTrafficControlScaffold(memo));
         SerialMonFrame f = new SerialMonFrame(memo);
-        // MonFrame needs a TrafficController for dispose() in line 51
-        try {
-            f.initComponents();
-        } catch (Exception ex) {
-            log.warn("SerialMonAction starting SerialMonFrame: Exception: {}", ex.toString());
-        }
+        // MonFrame needs a TrafficController for dispose()
+
+        f.initComponents();
+        
         f.pack();
         f.setVisible(true);
```

Comment:
Can you move these lines to the GUI thread? See issue #14688 and PR #14689.

Reply:
No problem @danielb987 .
Does https://www.jmri.org/help/en/html/doc/Technical/JUnit.shtml#testSwingCode need an update following these?
At present the advice is that if the JUnit thread is not interacting with the GUI thread ( as in this case ), it doesn't have to be on the GUI Thread.
I can see how updates to test code would be more consistent / reviewable if this advice was removed, thoughts ?

Reply:
After a bit more research, I've learned a little more about this.  There are changes between Java 21 and Java 25 that change the Swing threading advice:  _All_ Swing interactions, except those very few methods marked as thread safe, should be be done on the Swing/AWT thread.  That includes constructing GUIs before the `setVisible(true)` call.  So the advice on that page should change to recommend that Swing interactions be deferred to the Swing/AWT thread.  That's going to be a pain to do completely....